src/porepy/geometry/constrain_geometry.py

```diff
@@ -1,30 +1,44 @@
-b""" Module with various functions to constrain a geometry.
+""" Module with various functions to constrain a geometry.
 
 Examples are to cut objects to lie within other objects, etc.
 """
+from typing import Union
+
 import numpy as np
 
 import porepy as pp
 
 
-def lines_by_polygon(poly_pts, pts, edges):
+def lines_by_polygon(
+    poly_pts: np.ndarray, pts: np.ndarray, edges: np.ndarray
+) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
     """
     Compute the intersections between a polygon (also not convex) and a set of lines.
-    The computation is done line by line to avoid the splitting of edges caused by other
-    edges. The implementation assume that the polygon and lines are on the plane (x, y).
+
+    The computation is done line by line to avoid the splitting of edges caused by
+    other edges. The implementation assumes that the polygon and lines are on the plane
+    (x, y).
 
     Parameters:
-    poly_pts (np.ndarray, 3xn or 2xn): the points that define the polygon
-    pts (np.ndarray, 3xn or 2xn): the points associated to the lines
-    edges (np.ndarray, 2xn): for each column the id of the points for the line
+        poly_pts (shape=(nd, num_points)): points that define the polygon.
+        pts (shape=(nd, num_points)): points associated to the lines.
+        edges (shape=(2, num_points)): for each column the id of the points for the
+            line.
 
     Returns:
-    int_pts (np.ndarray, 2xn): the point associated to the lines after the intersection
-    int_edges (np.ndarray, 2xn): for each column the id of the points for the line after
-        the intersection. If the input edges have tags, stored in rows [2:], these will
-        be preserved.
-    edges_kept (np.ndarray, n): Column index of the kept edges. This will have recurring
-        values if an edge is cut by a non-convex domain.
+        A tuple with three elements.
+
+        ndarray ``(shape=(2, num_points))``:
+            Points associated to the lines after the intersection.
+
+        ndarray ``(shape=(2, num_points) dtype=int)``:
+            For each column the id of the points for the line after the intersection.
+            If the input edges have tags, stored in rows [2:], these will be
+            preserved.
+
+        ndarray ``(shape=(num_points, ), dtype=int)``:
```

Comment:
Maybe write len = num_points instead of shape=(num_points, )?

Reply:
I think in this case is rather important to simply avoid confusion and use shape. Especially, since this is an array of integers.